ACPs/99-validatorsetmanager-contract/README.md

```diff
@@ -0,0 +1,265 @@
+```text
+ACP: 99
+Title: ValidatorSetManager Solidity Contract
+Author(s): Gauthier Leonard <https://github.com/Nuttymoon>
+Discussions-To: https://github.com/avalanche-foundation/ACPs/discussions/98
+Status: Proposed
+Track: Best Practices
+```
+
+## Abstract
+
+Define (i) a reference implementation for a minimal Solidity smart contract to manage a Subnet’s validator set from an Avalanche EVM chain, as well as (ii) a reference architecture to easily plug in custom “security modules” on top of this contract (e.g. to implement a PoS Subnet).
+
+The proposed name for this contract is `ValidatorSetManager`.
+
+This proposal should also help define the payload of the P-Chain notifications sent to the Subnet upon validator addition and removal.
+
+This ACP relies on concepts introduced in [ACP-73 (Warp Addressed Transactions)](https://github.com/avalanche-foundation/ACPs/pull/73) and [ACP-77 (Reinventing Subnets)](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/77-reinventing-subnets). It depends on them to transition to an `Implementable` state.
+
+## Motivation
+
+[ACP-77 (Reinventing Subnets)](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/77-reinventing-subnets) opens the door to managing a Subnet validator set (stored on the P-Chain) from another chain of the Avalanche network. This is achieved by sending updates to the P-Chain from a “validator manager” address on this chain. The “validator manager” address will be the only one able to issue `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx` on the P-Chain to add, change the weight of, and remove validators.
+
+`Warp Addressed Transactions` (introduced in [ACP-73](https://github.com/avalanche-foundation/ACPs/pull/73)) could be used to send the validator set updates to the P-Chain.
```

Comment:
I think we should remove mention of ACP-73 since it won't be merged into the ACP repo (cc @aaronbuchwald to confirm)

Reply:
We talked about this and I said it was fine to include to show the progression like including a link to an article 🤷

Reply:
I talked with @Nuttymoon about this and I said it was fine to include to show the progression of ideas like referencing an article 🤷

Reply:
The mention of ACP-73 in the "Abstract" makes sense to me for that reason. This one makes it seem like this ACP relies on ACP-73 being activated with the "Warp Addressed Transactions" (I believe you pointed this out too here https://github.com/avalanche-foundation/ACPs/pull/99#discussion_r1628354647)

Reply:
I guess it is fine to remove ACP-73 references altogether as it seems that using the `AddessedCall` format is enough to issue transactions on the P-Chain.